memory/memory_index.py

```
@@ -0,0 +1,981 @@
+            rows = self._conn.execute("""
+                SELECT v.id, v.distance, m.*
+                FROM memories_vec v
+                JOIN memories m ON v.id = m.id
+                WHERE v.embedding MATCH ?
+                AND k = ?
+            """, (emb_json, k * 3)).fetchall()
+
+            results = []
+            for row in rows:
+                d = self._row_to_dict(row)
+                # 距离转相似度（余弦距离）
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)